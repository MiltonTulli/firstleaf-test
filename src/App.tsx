import React from 'react';
import { Switch, Route, Redirect } from 'react-router-dom';
import { Countries } from './containers';
import { Landing, NotFound } from './components';
import R from './utils/routes';

function App() {
  return (
    <Switch>
      <Route exact path={R.ROOT} component={Landing}></Route>
      <Route path={R.APP} component={Countries} />
      <Route path={R.NOT_FOUND} component={NotFound} />
      <Redirect to={R.NOT_FOUND} />
    </Switch>
  );
}

export default App;
