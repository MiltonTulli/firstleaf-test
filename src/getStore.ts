/* istanbul ignore file */
import { createStore, applyMiddleware } from 'redux';
import { createLogger } from 'redux-logger';
import thunk from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension';
import reducer, { initialState } from './modules';

export default () => {
  const middlewareChain = [thunk];
  if (process.env.NODE_ENV === 'development') {
    const logger = createLogger();
    middlewareChain.push(logger);
  }
  const store = createStore(
    reducer,
    initialState,
    composeWithDevTools(applyMiddleware(...middlewareChain))
  );
  return store;
};
