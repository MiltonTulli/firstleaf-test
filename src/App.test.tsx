import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import App from './App';
import getStore from './getStore';

describe('<App>', () => {
  let store;

  beforeEach(() => {
    store = getStore();
  });

  it('Should match snapshot', () => {
    const rendered = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <App />
          </BrowserRouter>
        </Provider>
      )
      .toJSON();
    expect(rendered).toMatchSnapshot();
  });
});
