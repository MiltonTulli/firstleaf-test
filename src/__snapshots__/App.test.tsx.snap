// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App> Should match snapshot 1`] = `
<div
  className="makeStyles-root-1"
>
  <h3
    className="MuiTypography-root makeStyles-header-2 MuiTypography-h3"
  >
    FirstLeaf Test
  </h3>
  <a
    className="makeStyles-link-3"
    href="/app"
    onClick={[Function]}
  >
    <h6
      className="MuiTypography-root makeStyles-header-2 MuiTypography-h6"
    >
      Start
    </h6>
  </a>
</div>
`;
